### ISSUE TITLE
Short-circuit operator?

### ISSUE THREAD

**Issue by evanmiller** [ggerganov/ggml#357]
I'm playing with an idea to short-circuit a graph if a certain condition is met. Specifically, I'd like to have one or more "checkpoints" in a Transformer model where the logits are evaluated mid-graph – if they overwhelmingly favor one token, the remaining layers are ignored. Is a short-circuit functionality something that would be considered in GGML? A very basic version could just abort the graph computation; a more complicated one might point to a downstream task.

**Comment by evanmiller** [comment]
Related: #283

**Comment by ggerganov** [comment]
I think you can achieve that with #308 and a custom op defined in user code.
When you build the graph, you add the custom op at the points where you are interested to have it and the op would set a `bool` for example if the "short-circuit" condition is met. The abort callback provided to `ggml_graph_compute()` would check the `bool` and abort the computation if is set
